penguin/tools/action_registry.py

```
@@ -91,6 +93,29 @@ def _write_file_input(params: Any) -> dict[str, Any]:
     }
 
 
+def _edit_file_input(params: Any) -> dict[str, Any]:
```

Ruff (0.15.12)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `params`

(ANN401)